src/vm/dllimportcallback.cpp

```diff
@@ -1412,6 +1407,9 @@ VOID UMThunkMarshInfo::RunTimeInit()
         new (&sigInfo) PInvokeStaticSigInfo(pMD);
     else
         new (&sigInfo) PInvokeStaticSigInfo(GetSignature(), GetModule());
+
+    m_cbActualArgSize = (pStubMD != NULL) ? pStubMD->AsDynamicMethodDesc()->GetNativeStackArgSize()
```

Comment:
You should keep track of the size in the loop above. It is wasteful to get the size by calling these methods - MethodDesc::SizeOfArgStack runs the exact same loop over all arguments.